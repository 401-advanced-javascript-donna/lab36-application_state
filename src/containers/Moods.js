import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Controls from '../components/controls/Controls';
import Face from '../components/face/Face';
import { getActions, getFace } from '../selectors/moodSelector';


const Moods = ({ actions, emoji, handleSelection }) => {
  return (
    <>
      <Controls actions={actions} handleSelection={handleSelection} />
      <Face emoji={emoji} />
    </>
  );
};

Moods.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.object).isRequired,
  emoji: PropTypes.string.isRequired,
  handleSelection: PropTypes.func.isRequired
};



const mapStateToProps = (state) => ({
  emoji: getFace(state),
  actions: getActions(state)
});

const mapDispatchToProps = dispatch => ({
  handleSelection(name) {
    dispatch({
      type: name
    });
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Moods);
