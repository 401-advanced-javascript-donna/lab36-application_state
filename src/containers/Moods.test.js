import React from 'react';
import { shallow } from 'enzyme';
import store from '../store';
import Moods from './Moods';

describe('Moods', () => {
  describe('container', () => {
    let wrapper;
    store.dispatch = jest.fn();
    beforeEach(() => {
      wrapper = shallow(<Moods store={store} />).dive();
    });

    it('matches a snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('updates state on DRINK_COFFEE selection', () => {
      wrapper.props().handleSelection('DRINK_COFFEE');

      expect(store.dispatch).toHaveBeenCalledWith({
        type: 'DRINK_COFFEE'
      });
    });

    it('updates state on TAKE_NAP selection', () => {
      wrapper.props().handleSelection('TAKE_NAP');

      expect(store.dispatch).toHaveBeenCalledWith({
        type: 'TAKE_NAP'
      });
    });

    it('updates state on EAT_SNACK selection', () => {
      wrapper.props().handleSelection('EAT_SNACK');

      expect(store.dispatch).toHaveBeenCalledWith({
        type: 'EAT_SNACK'
      });
    });

    it('updates state on STUDY selection', () => {
      wrapper.props().handleSelection('STUDY');

      expect(store.dispatch).toHaveBeenCalledWith({
        type: 'STUDY'
      });
    });
  });
});
