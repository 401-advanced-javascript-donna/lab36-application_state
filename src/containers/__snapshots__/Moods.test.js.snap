// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moods container matches a snapshot 1`] = `
<Moods
  actions={
    Array [
      Object {
        "count": 0,
        "name": "DRINK_COFFEE",
        "text": "Drink Coffee",
      },
      Object {
        "count": 0,
        "name": "EAT_SNACK",
        "text": "Snack",
      },
      Object {
        "count": 0,
        "name": "TAKE_NAP",
        "text": "Nap",
      },
      Object {
        "count": 0,
        "name": "STUDY",
        "text": "Study",
      },
      Object {
        "name": "RESET",
        "text": "Reset",
      },
    ]
  }
  emoji="ðŸ¤¬"
  handleSave={[Function]}
  handleSelection={[Function]}
  histories={Array []}
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
